dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -2383,6 +2387,78 @@ public Set<OrcidAuthorInformation> getWorkflowVersionOrcidAuthors(@Parameter(hid
         return orcidAuthorInfo;
     }
 
+    @GET
+    @RolesAllowed({"admin", "curator"})
+    @Path("/versionsNeedingRetroactiveDoi")
+    @UnitOfWork
+    @Timed
+    @Operation(operationId = "getVersionsNeedingRetroactiveDoi", description = "Calculates a list of workflow versions that need a retroactive DOI", security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME))
+    @SuppressWarnings("checkstyle:MagicNumber")
+    public List<WorkflowAndVersion> getVersionsNeedingRetroactiveDoi(@Parameter(hidden = true) @Auth User user,
+        @QueryParam("limit") @Min(1) @Max(1000) @DefaultValue("100") Integer limit) {
+        // Retrieve the information we'll use to select the workflows "most eligible" for a retroactive DOI.
+        Map<Long, Long> workflowIdToDoiCount = workflowDAO.getWorkflowIdsAndDoiCounts();
+        Set<Long> eligibleWorkflowIds = workflowDAO.getWorkflowIdsEligibleForRetroactiveDoi();
+        Set<Long> gitHubDoiWorkflowIds = workflowDAO.getWorkflowIdsWithGitHubDoi();
+
+        // Determine the workflows "most eligible" for a DOI, which are the workflows that don't have a GitHub DOI
```

Comment:
question: should we rule out workflows with a user generated DOI as well as those that were issued more directly on GitHub?

Reply:
I feel somewhat strongly that if a user has went to the trouble of linking their accounts, persisting through to manual DOIs, maybe they do want to handle/control the process. 
i.e. maybe that "one" DOI they generated is really the one they want people to focus on

In other words, I was a bit surprised because this PR rules out direct GitHub but not manual DOIs whereas I would have kinda ruled out manual DOIs but less sure about direct GitHub.

Reply:
Per our slack convo https://ucsc-gi.slack.com/archives/C05EZH3RVNY/p1742244954907069, I changed the endpoint to exclude workflows that have at least one GitHub OR manual DOI.